war/src/main/java/io/lumeer/engine/controller/SecurityFacade.java

```diff
@@ -342,4 +345,53 @@ private String userOrGroupKey(String user) {
    private String userOrGroupName(String user, String group) {
       return user != null ? user : group;
    }
+   /********* GETTING ROLES *********/
+   public List<Role> getOrganizationRoles(String organizationCode) {
+      DataDocument doc = systemDataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ORGANIZATION_ROLES_COLLECTION_NAME,
+            dataStorageDialect.fieldValueFilter(LumeerConst.Security.ORGANIZATION_ID_KEY,
+                  organizationFacade.getOrganizationId(organizationCode)),
+            Arrays.asList(LumeerConst.Security.ROLES_KEY));
+
+      return getRolesList(doc);
+   }
+
+   public List<Role> getProjectRoles(String projectCode) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(projectFilter(projectCode)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY)));
+
+      return getRolesList(doc);
+   }
+
+   public List<Role> getCollectionRoles(String projectCode, String collectionName) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(collectionFilter(projectCode, collectionName)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY)));
+
+      return getRolesList(doc);
+   }
+
+   public List<Role> getViewRoles(String projectCode, int viewId) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(viewFilter(projectCode, viewId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY)));
+
+      return getRolesList(doc);
+   }
+
+   private List<Role> getRolesList(DataDocument doc) {
+      List<Role> output = new ArrayList<>();
+      doc.getDataDocument(LumeerConst.Security.ROLES_KEY).forEach((k, v)-> {
+         output.add(new Role(
+               k,
+               ((Map<String, List<String>>) v).get(LumeerConst.Security.USERS_KEY),
+               ((Map<String, List<String>>) v).get(LumeerConst.Security.GROUP_KEY)));
```

Comment:
`Role` class should have a constructor that takes `DataDocument` as a parameter in order to make this method simpler.